src/main/java/com/paneedah/mwc/groovyscript/script/CraftingStation.java

```diff
@@ -0,0 +1,177 @@
+
+public class CraftingStation extends VirtualizedRegistry<GSCrafting> {
+
+    @Override
+    public void onReload() {
+        this.removeScripted().forEach(CraftingRegistry::deleteRecipeRegistry);
+        this.restoreFromBackup().forEach(CraftingRegistry::registerRecipe);
+    }
+
+
+    //Nuke ALL recipes. no matter the category.
+
+    public void removeAll(){
+        clearRegistry();
+    }
+
+    //Remove Recipe that outputs given Ingredient.
+
+    public void remove(IIngredient ingredient){
+        for(ArrayList<IModernCrafting> list:craftingMap.values()){
+            for(IModernCrafting recipe: list){
+                if(ingredient.test(new ItemStack(recipe.getItem()))){
+                    deleteRecipeRegistry(recipe);
+                }
+            }
+        }
+    }
+
+    //Nuke ALL recipes of Given Category.
+
+    public void removeAllinGroup(CraftingGroup group){
+        clearGroup(group);
+    }
+
+    public void removeAllinGroup(String group){
+        clearGroup(CraftingGroup.valueOf(group));
+    }
+
+    public void removeinGroup(IIngredient ingredient, CraftingGroup group) {
+        for (IModernCrafting recipe : craftingMap.get(group)) {
+            if (ingredient.test(new ItemStack(recipe.getItem()))) {
+                deleteRecipeRegistry(recipe);
+            }
+        }
+    }
+
+    public void removeinGroup(IIngredient ingredient, String group) {
+        removeinGroup(ingredient, CraftingGroup.valueOf(group));
+    }
+
+    public void addRecipe(IModernCrafting crafting){
+        registerRecipe(crafting);
+    }
+
+    //Start Recipe Builder
+
+    public RecipeBuilder recipeBuilder(){
+        return new RecipeBuilder();
+    }
+
+    public static class RecipeBuilder extends AbstractRecipeBuilder<GSCrafting>{
+
+        @Override
+        public String getErrorMsg() {
+            return "Error adding Workbench Recipe";
+        }
+
+        private final ArrayList<CraftingEntry> entries = new ArrayList<>();
+        private CraftingGroup group = GUN;
+
+        //Add Ingredient of recipe With ItemStack
+
+        public RecipeBuilder addEntry(ItemStack stack){
+            return addEntry(stack, 1);
+        }
+
+        //Add Ingredient of recipe With Ore Dictionary Value
+
+        public RecipeBuilder addOredict(String entry, int count){
+            return addOredict(entry, count, 1);
+        }
+
+        //Add Ingredient of recipe With given ItemStack and yield(0.0~1.0)
+
+        public RecipeBuilder addEntry(ItemStack stack, double yield){
+            this.entries.add(new CraftingEntry(stack.getItem(), stack.getCount(), yield));
+            return this;
+        }
+
+        //Add Ingredient of recipe With given Ore Dictionary Entry and yield(0.0~1.0)
+
+
+        public RecipeBuilder addOredict(String entry, int count, double yield){
+            this.entries.add(new CraftingEntry(OreDictionary.getOres(entry).get(0).getItem(),entry, count, yield));
+            return this;
```

Comment:
I recommend to just use the input method of the super class. For the yield simply add a `DoubleList` and override the `input` methods to add a default yield. In the `register` method you then parse the `IIngredient` with instance checks.